shared/login/register/code-page/index.render.native.js

```diff
@@ -8,222 +8,223 @@ import Container from '../../forms/container'
     return (
-      <Box style={stylesIntro}>
-        <Text type='Header' style={{marginBottom: 10}}>Enter text code</Text>
-        <Text type='BodySmall'>In the Keybase App</Text>
-        <Text type='BodySmall'>{'go to Devices > Add a new device'}</Text>
+      <Box style={stylesEnterText}>
+        <Icon type='icon-phone-text-code-32' style={{alignSelf: 'center'}} />
+        <Input
+          hintText='opp blezzard tofi pando agg whi pany yaga jocket daubt bruwnstane hubit yas'
+          floatingHintTextOverride='Text code'
+          multiline={true}
+          rowsMin={3}
+          value={this.props.enterText}
+          onChangeText={text => this.props.onChangeText(text)}
+        />
+        <Button type='Primary' style={{marginTop: 10, marginBottom: 20}} label='Continue' onClick={() => this.props.textEntered(codePageModeEnterText)} />
       </Box>
     )
   }
 
-  renderIntroScanCode () {
+  renderSwitchButton (mode: Mode, icon: IconType, label: string) {
     return (
-      <Box style={stylesIntro}>
-        <Text type='Header' style={{marginBottom: 10}}>Scan QR code</Text>
-        <Text type='BodySmall'>In the Keybase App</Text>
-        <Text type='BodySmall'>{'go to Devices > Add a new device'}</Text>
-      </Box>
+      <ClickableBox
+        underlayColor={globalColors.white}
+        onClick={() => this.props.setCodePageMode(mode)}
+        style={{marginBottom: 20}}>
+        <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', alignSelf: 'center', marginLeft: 10, marginRight: 10}}>
+          <Icon type={icon} />
+          <Text type='BodyBigLink' style={{marginLeft: 4}}>{label}</Text>
+        </Box>
+      </ClickableBox>
     )
   }
 
-  renderIntroShowQR () {
+  renderScanCodeForDesktop () {
     return (
-      <Box style={stylesIntro}>
-        <Text type='Header' style={{marginBottom: 10}}>Scan this QR code</Text>
-        <Text type='BodySmall'>{'When adding a new mobile device'}</Text>
-      </Box>
+      <Container style={stylesContainer} onBack={this.props.onBack}>
+        <Box style={stylesIntro}>
+          <Text type='Header' style={{marginBottom: 10}}>Scan QR code</Text>
+          <Text type='Body'>In the Keybase app on your computer,</Text>
+          <Text type='Body'>{'go to Devices > Add a new device.'}</Text>
+        </Box>
+        {this.renderScanCode()}
+        {this.renderSwitchButton(codePageModeEnterText, 'icon-phone-text-code-32', 'Type text code instead')}
+      </Container>
     )
   }
 
-  renderIntro () {
-    const headerTextMap: {[key: Mode]: React$Element<*>} = {
-      codePageModeScanCode: this.renderIntroScanCode(),
-      codePageModeShowCode: this.renderIntroShowQR(),
-      codePageModeShowText: this.renderIntroShowTextCode(),
-      codePageModeEnterText: this.renderIntroEnterTextCode(),
-    }
-
-    return headerTextMap[this.props.mode] || <Text type='BodySmall'>Good luck</Text>
-  }
-
-  renderShowText () {
+  renderShowCodeForDesktop () {
```

Comment:
I'm not sure... maybe you could show QR code to desktop via it's camera... Like put your phone up to the macbook camera.

I don't think we do that yet but, for completeness, I included it.